include/alpaka/core/RuntimeMacros.hpp

```diff
@@ -0,0 +1,39 @@
+/* Copyright 2022  Andrea Bocci, Mehmet Yusufoglu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+//! ALPAKA_DEVICE_THROW either aborts(terminating the program and creating a core dump) or throws std::runtime_error
+//! depending on the Acc. The std::runtime_error exception can be catched in the catch block.
+//!
+//! CUDA __trap function triggers std::runtime_error but can be catched during wait not exec.
+//!
+//! The OpenMP specification mandates that exceptions thrown by some thread must be handled by the same thread.
+//! Therefore std::runtime_error thrown by ALPAKA_DEVICE_THROW aborts the the program for OpenMP backends. If needed
+//! the SIGABRT signal can be catched by signal handler.
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && defined(__CUDA_ARCH__)
+#    define ALPAKA_DEVICE_THROW(MSG)                                                                                  \
+        {                                                                                                             \
+            printf("%s [ALPAKA_DEVICE_THROW: Calling __trap(). Cuda backend is enabled.]\n", (MSG));                  \
```

Comment:
I think a message like
```
%s [ALPAKA_DEVICE_THROW: Calling __trap(). Cuda backend is enabled.]\n
```
is fine for debugging the implementation - but once this goes in production can we print just the user-supplied message ?

Or, if we do want to add more information, something like
```
alpaka encountered a user-defined error condition while running on the ___ back-end:\n
%s
```

Reply:
ok, changed. Thanks.